tests/test_end_to_end.py

```diff
@@ -0,0 +1,95 @@
+"""
+End-to-end tests for each backend configuration. Sets up a server with
+the backend, sends some basic queries to that server and verifies results
+are as expected.
+"""
+import unittest
+
+from ga4gh.backends import WormtableBackend
+from tests.test_backends import WormtableTestFixture
+import ga4gh.protocol as protocol
+import ga4gh.server as server
+
+
+class EndToEndTest(unittest.TestCase):
+    def setUpServer(self, backend):
+        server.app.config['TESTING'] = True
+        server.app.config["VariantBackend"] = backend
+        self.app = server.app.test_client()
+
+    def sendJSONPostRequest(self, path, data):
+        return self.app.post(path,
+                             headers={'Content-type': 'application/json'},
+                             data=data)
+
+    def runVariantSetsSearch(self):
+        expectedIds = ['example_1', 'example_2', 'example_3', 'example_4']
+        request = protocol.GASearchVariantSetsRequest()
+        response = self.sendJSONPostRequest('/variantsets/search',
+                                            request.toJSONString())
+
+        self.assertEqual(200, response.status_code)
+
+        responseData = protocol.GASearchVariantSetsResponse.fromJSON(
+            response.data)
+        self.assertTrue(protocol.GASearchVariantSetsResponse.validate(
+            responseData.toJSONDict()))
+
+        self.assertIsNone(responseData.nextPageToken)
+        self.assertEqual(4, len(responseData.variantSets))
+        for variantSet in responseData.variantSets:
+            self.assertTrue(variantSet.id in expectedIds)
+
+    def runVariantsSearch(self):
+        expectedIds = ['example_1']
```

Comment:
Please insert a line `referenceName = '1'` and replace the hard-coded values below. Also add a TODO comment stating we should get these values from the test backend when it's being refactored.